packages/ui/src/components/Checkbox.tsx

```diff
@@ -1 +1,44 @@
-export { type CheckboxProps, Checkbox } from 'theme-ui';
+import * as React from 'react';
+import {
+  Checkbox as CheckBoxUi,
+  CheckboxProps as CheckboxPropsUi,
+} from 'theme-ui';
+
+export interface CheckboxProps extends CheckboxPropsUi {
+  indeterminate?: boolean;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  className?: string;
+}
+
+export const Checkbox: React.FC<CheckboxProps> = ({
+  indeterminate = false,
+  className = '',
+  onChange,
+  checked,
+  ...rest
+}) => {
+  const ref = React.useRef<HTMLInputElement>(null);
+
+  React.useEffect(() => {
+    if (ref.current) {
+      ref.current.indeterminate = indeterminate;
+    }
+  }, [indeterminate]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (onChange) {
+      onChange(event);
+    }
+  };
+
+  return (
+    <CheckBoxUi
+      type="checkbox"
+      ref={ref}
+      onChange={handleChange}
+      checked={checked}
```

Comment:
Why do we need `checked` on props if we've already provided `...rest`?

Reply:
i define this in tests from input as checkbox, but i removed